pkg/domain/infra/abi/images.go

```diff
@@ -747,28 +736,27 @@ func (ir *ImageEngine) Sign(ctx context.Context, names []string, options entitie
 					}
 				}
 			}
+			sigStoreDir, err = localPathFromURI(sigStoreDir)
+			if err != nil {
+				return nil, errors.Wrapf(err, "invalid signature storage %s", sigStoreDir)
+			}
 		}
-		sigStoreDir, err = isValidSigStoreDir(sigStoreDir)
-		if err != nil {
-			return nil, errors.Wrapf(err, "invalid signature storage %s", sigStoreDir)
-		}
-		repos, err := newImage.RepoDigests()
+		manifestDigest, err := manifest.Digest(getManifest)
 		if err != nil {
-			return nil, errors.Wrapf(err, "error calculating repo digests for %s", signimage)
+			return nil, err
 		}
-		if len(repos) == 0 {
-			logrus.Errorf("no repodigests associated with the image %s", signimage)
-			continue
+		repo := reference.Path(dockerReference)
+		if path.Clean(repo) != repo { // Coverage: This should not be reachable because /./ and /../ components are not valid in docker references
+			return nil, errors.Errorf("Unexpected path elements in Docker reference %s for signature storage", dockerReference.String())
 		}
 
 		// create signature
 		newSig, err := signature.SignDockerManifest(getManifest, dockerReference.String(), mech, options.SignBy)
 		if err != nil {
 			return nil, errors.Wrapf(err, "error creating new signature")
 		}
-
-		trimmedDigest := strings.TrimPrefix(repos[0], strings.Split(repos[0], "/")[0])
-		sigStoreDir = filepath.Join(sigStoreDir, strings.Replace(trimmedDigest, ":", "=", 1))
+		// create the signstore file
+		sigStoreDir = fmt.Sprintf("%s@%s=%s", filepath.Join(sigStoreDir, repo), manifestDigest.Algorithm(), manifestDigest.Hex())
```

Comment:
(Consider introducing a separate `signatureDir` here, again to avoid overloading a single variable with multiple meanings.)